### PR TITLE
Feat/148

### PR DESCRIPTION
**Pull request by rizzom5000**
Converts team-based permissions to collaborator-based. These changes are dependent on the Github v3 API. 

A team view is kept in this version but is GET only.

### CODE REVIEW

File: documents/API.md [hunk 1]
```diff
@@ -4,38 +4,43 @@ This is a draft of the operations available to meet the initial views of the app
 
 ##Operations
 
-1. Get list of users. If permission_repo is specified, the "permission" flag on each user will be present, reporting their access to that repo.
+1. Get identity of authenticated user
+  
+        GET /identity
+        returns Identity{}
+
+2. Get list of users. If permission_repo is specified, the "permission" flag on each user will be present, reporting their access to that repo.
```
Comment (natoverse) [inline documents/API.md]: You can use '1.' repeatedly in markdown and it will auto-increment the numbers in the list.

**Comment by natoverse** [comment]
Did you have to add a customer header to use the new collaborators functionality?

File: app/server/components/repositories/util/permissions.js [hunk 2]
```diff
@@ -1,96 +1,34 @@
...
+        return new Promise((resolve, reject) => {
+            Bluebird.join(userUtil.getGithubMembers(), userUtil.getGithubOwners(), repoUtil.getReposWithCollaborators(), (members, owners, repos) => {
+                let users = members.concat(owners);
+                orgMap.users = users.reduce((users, user) => {
+                    users[user.username] = user;
+                    return users;
+                }, {});
+                orgMap.repos = repos;
+                resolve(orgMap);
             });
```
Comment (natoverse) [inline app/server/components/repositories/util/permissions.js]: ooooh, nice.

File: app/server/components/services/github.mock.js [hunk 3]
```diff
@@ -91,6 +92,7 @@ var mask = require('json-mask'),
         '2': {
             id: 2,
             name: 'zzz-permissive-repo-Test-Repo-1-pull',
+            description: 'zzz-permissive-repo-Test-Repo-1-pull',
```
Comment (natoverse) [inline app/server/components/services/github.mock.js]: should these zzz teams be replaced entirely with "realistic" teams so it isn't confusing once they're all gone?

**Comment by rizzom5000** [comment]
@natoverse For now the custom request headers are still required. I've set this up so that it's configurable: "acceptHeader": "application/vnd.github.ironman-preview+json"

**Pull request merged**